### PR TITLE
Implement caching in the static API

### LINKED ISSUE #164: Cache languages and maps in the static api
Some methods in the static api are not using the cache:

  - [Versions](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/StaticRiotApi.cs#L945)
  - [Realm](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/StaticRiotApi.cs#L616)
  - [Languages](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/StaticRiotApi.cs#L406)
  - [LanguageStrings](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/StaticRiotApi.cs#L372)
  - [Maps](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/StaticRiotApi.cs#L432)

It'd be great to make them use the cache like the other methods in the static api.

### PR DESCRIPTION
**Pull request by shrayasr**
- [x] Versions
- [X] Realm
- [x] Languages
- [x] LanguageStrings
- [x] Maps

**PR pending review**

---
Fixes #164

### CODE REVIEW

**Comment by shrayasr** [comment]
`Maps` appears to be a wee bit more complex. Turns out that it sends back a far different type of response than the others. All other responses map well on to the `*StaticWrapper` -> `*Static` classes but for `Maps`. 

Here's my suggestion to keep it that way:

- `MapsStaticWrapper` has an instance of `MapsStatic`
- `MapsStatic` contains a `Dictionary<int, MapStatic> Data` field

That way we'll be able to keep the `GetMaps` and `GetMapsAsync` methods very much like the others.

Thoughts?

**Comment by BenFradet** [comment]
LGTM so far.

Your idea regarding the maps endpoint seems good, I'll review more carefully once it's implemented.

**Comment by shrayasr** [comment]
@BenFradet Ready for review, m'kind sir.

If you'd prefer them as 1 commit, post review I can squash them up.

File: RiotSharp/StaticRiotApi.cs [hunk 1]
```diff
@@ -21,16 +21,21 @@ public class StaticRiotApi : IStaticRiotApi
...
 
         private const string LanguageStringsRootUrl = "/api/lol/static-data/{0}/v1.2/language-strings";
+        private const string LanguageStringsCacheKey = "languagestrings";
 
         private const string LanguagesRootUrl = "/api/lol/static-data/{0}/v1.2/languages";
+        private const string LanguagesCacheKey = "languages";
 
         private const string MapRootUrl = "/api/lol/static-data/{0}/v1.2/map";
+        private const string MapCacheKey = "map";
+
```
Comment (BenFradet) [inline RiotSharp/StaticRiotApi.cs]: it seems there is an empty line too many

File: RiotSharp/StaticRiotApi.cs [hunk 2]
```diff
@@ -376,15 +382,25 @@ private StaticRiotApi(string apiKey)
...
             var json = requester.CreateGetRequest(string.Format(LanguageStringsRootUrl, region.ToString()), RootDomain,
                 new List<string> {
                     string.Format("locale={0}", language.ToString()),
                     string.Format("version={0}", version)
                 });
-            return JsonConvert.DeserializeObject<LanguageStringsData>(json);
+            var languageStrings = JsonConvert.DeserializeObject<LanguageStringsStatic>(json);
+
+            cache.Add(LanguageStringsCacheKey, new LanguageStringsStaticWrapper(languageStrings, language, version), DefaultSlidingExpiry);
```
Comment (BenFradet) [inline RiotSharp/StaticRiotApi.cs]: we usually try to keep lines that are smaller than 100 chars

File: RiotSharp/StaticRiotApi.cs [hunk 3]
```diff
@@ -394,15 +410,25 @@ private StaticRiotApi(string apiKey)
...
+
             var json = await requester.CreateGetRequestAsync(string.Format(LanguageStringsRootUrl, region.ToString()),
                 RootDomain, new List<string> {
                     string.Format("locale={0}", language.ToString()),
                     string.Format("version={0}", version)
                 });
-            return await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<LanguageStringsData>(json));
+            var languageStrings = await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<LanguageStringsStatic>(json));
+
+            cache.Add(LanguageStringsCacheKey, new LanguageStringsStaticWrapper(languageStrings, language, version), DefaultSlidingExpiry);
```
Comment (BenFradet) [inline RiotSharp/StaticRiotApi.cs]: same thing here

**Comment by BenFradet** [comment]
LGTM except for a few minor formatting quirks

**Comment by shrayasr** [comment]
Fixed up all the formatting quirks, sorry about that :smile: :+1:

File: RiotSharp/StaticRiotApi.cs [hunk 4]
```diff
@@ -41,6 +45,7 @@ public class StaticRiotApi : IStaticRiotApi
         private const string SummonerSpellCacheKey = "spell";
 
         private const string VersionRootUrl = "/api/lol/static-data/{0}/v1.2/versions";
+        private const string VersionCacheKey = "version";
```
Comment (BenFradet) [inline RiotSharp/StaticRiotApi.cs]: same, this should be `versions`

File: RiotSharp/StaticRiotApi.cs [hunk 5]
```diff
@@ -394,15 +410,26 @@ private StaticRiotApi(string apiKey)
...
+                return wrapper.LanguageStringsStatic;
+            }
+
             var json = await requester.CreateGetRequestAsync(string.Format(LanguageStringsRootUrl, region.ToString()),
                 RootDomain, new List<string> {
                     string.Format("locale={0}", language.ToString()),
                     string.Format("version={0}", version)
                 });
-            return await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<LanguageStringsData>(json));
+            var languageStrings = await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<LanguageStringsStatic>(json));
```
Comment (BenFradet) [inline RiotSharp/StaticRiotApi.cs]: this line seems a bit long to me, are you sure it's less than 100 chars?

**Comment by BenFradet** [comment]
No problem, there is still one minor issue regarding cache key naming and another regarding formatting and I'll be ready to merge!

**Comment by shrayasr** [comment]
All fixed up for you, kind sir :ok_hand:

**Comment by BenFradet** [comment]
Merging, thanks a lot!

**Pull request merged**